app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarView.kt

```diff
@@ -109,34 +109,40 @@ class BrowserToolbarView(
             view.apply {
                 elevation = TOOLBAR_ELEVATION.dpToFloat(resources.displayMetrics)
 
-                onUrlClicked = {
+                if (!isCustomTabSession) {
+                    display.setUrlBackground(getDrawable(R.drawable.search_url_background))
+                }
+
+                display.onUrlClicked = {
                     interactor.onBrowserToolbarClicked()
                     false
                 }
 
-                browserActionMargin = browserActionMarginDp.dpToPx(resources.displayMetrics)
-
-                urlBoxView = if (isCustomTabSession) null else urlBackground
-                progressBarGravity = if (isCustomTabSession) PROGRESS_BOTTOM else PROGRESS_TOP
-
-                textColor = ContextCompat.getColor(context, R.color.photonGrey30)
-
-                hint = context.getString(R.string.search_hint)
-
-                suggestionBackgroundColor = ContextCompat.getColor(
-                    container.context,
-                    R.color.suggestion_highlight_color
-                )
+                display.progressGravity = if (isCustomTabSession) {
+                    DisplayToolbar.Gravity.TOP
+                } else {
+                    DisplayToolbar.Gravity.BOTTOM
+                }
 
-                textColor = ContextCompat.getColor(
+                val primaryTextColor = ContextCompat.getColor(
                     container.context,
                     ThemeManager.resolveAttribute(R.attr.primaryText, container.context)
                 )
-
-                hintColor = ContextCompat.getColor(
+                val secondaryTextColor = ContextCompat.getColor(
                     container.context,
                     ThemeManager.resolveAttribute(R.attr.secondaryText, container.context)
                 )
+
+                display.colors = display.colors.copy(
+                    text = primaryTextColor,
+                    securityIconSecure = primaryTextColor,
+                    securityIconInsecure = primaryTextColor,
+                    menu = primaryTextColor,
+                    hint = secondaryTextColor,
+                    trackingProtection = primaryTextColor
```

Comment:
We also have to indicate the `separator` color  it should be `?toolbarDivider`

Reply:
Thanks! Updated. :)